files/image_config/monit/proc_stat

```diff
@@ -0,0 +1,122 @@
+#
+  set daemon 60            # check services at 1-minute intervals
+#   with start delay 240    # optional: delay the first check by 4-minutes (by
+#                           # default Monit check immediately after Monit start)
+set logfile syslog
+set idfile /var/lib/monit/id
+#
+###############################################################################
+## Includes all the critical processes running in SONiC containers.
+###############################################################################
+
+# For syncd container.
+check process syncd matching "/usr/bin/syncd --diag"
```

Comment:
This file will reside in the host OS, but all of the processes you're monitoring reside in containers. To align with the containerized nature of SONiC, it makes more sense to have container-specific monit configuration reside with the container-specific files, thus the host OS configuration can be ignorant of any and all containers.

I suggest adding a file in each container which specifies the processes to monitor which run _in that container_, then use our existing mechanism to copy that file to the base image during image compilation. If you copy all the files to /etc/monit/conf.d/, as long as monitrc is configured to load from this directory (it is by default -- see [here](https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/monit/monitrc#L294)), it will load all of those config files when it starts up.

Reply:
Good suggestion! I got your point! Will do it. Thanks.